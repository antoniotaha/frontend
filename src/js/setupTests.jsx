import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import httpAdapter from 'axios/lib/adapters/http'
import axios from 'axios'

export function shallowRenderAndMatch(componentJsx) {
	const wrapper = shallow(componentJsx)
	expect(wrapper).toMatchSnapshot()
	return wrapper
}

Enzyme.configure({ adapter: new Adapter() })
axios.defaults.adapter = httpAdapter 
