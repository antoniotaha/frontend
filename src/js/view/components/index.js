
export * from './elements/Button'
export * from './elements/Card'
export * from './elements/Image'
export * from './elements/Input'
export * from './elements/Label'
export * from './elements/Pagination'
export * from './elements/TextArea'
export * from './elements/TextInput'
export * from './elements/SuccessModal'


export * from './form/Control'
export * from './form/Field'

export * from './grid/Column'
export * from './grid/Columns'

export * from './layout/Header'
export * from './layout/Conteudo'

