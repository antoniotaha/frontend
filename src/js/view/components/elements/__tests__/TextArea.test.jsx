import * as React from 'react'
import * as C from 'view/components'
import { shallowRenderAndMatch } from 'setupTests'

describe('TextArea', () => {
	it('TextArea sem label', function () {
		shallowRenderAndMatch(<C.TextArea />)
	})

	it('TextArea com label', function () {
		shallowRenderAndMatch(<C.TextArea label='Teste' placeholder='Teste' name='teste' />)
	})
})
