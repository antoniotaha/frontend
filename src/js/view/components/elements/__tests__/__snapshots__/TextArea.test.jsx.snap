// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea TextArea com label 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextArea
    label="Teste"
    name="teste"
    placeholder="Teste"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <textarea
        className="textarea"
        label="Teste"
        name="teste"
        placeholder="Teste"
/>,
      "label": "Teste",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "className": "textarea",
        "label": "Teste",
        "name": "teste",
        "placeholder": "Teste",
      },
      "ref": null,
      "rendered": null,
      "type": "textarea",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <textarea
          className="textarea"
          label="Teste"
          name="teste"
          placeholder="Teste"
/>,
        "label": "Teste",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "textarea",
          "label": "Teste",
          "name": "teste",
          "placeholder": "Teste",
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`TextArea TextArea sem label 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextArea />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <textarea
        className="textarea"
        name={undefined}
        placeholder={undefined}
/>,
      "label": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "className": "textarea",
        "name": undefined,
        "placeholder": undefined,
      },
      "ref": null,
      "rendered": null,
      "type": "textarea",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <textarea
          className="textarea"
          name={undefined}
          placeholder={undefined}
/>,
        "label": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "className": "textarea",
          "name": undefined,
          "placeholder": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "textarea",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
